inc/taxonomy-content-language.php

```diff
@@ -74,6 +84,14 @@ function wmf_get_and_maybe_create_current_language_term(): ?WP_Term {
  * @return \WP_Term|null
  */
 function wmf_create_current_language_term(): ?WP_Term {
+
+	// When this function is called from the REST API, the admin taxonomy
+	// utilities may not be available. This fix ensures that the content
+	// language functionality works in the block editor.
+	if ( ! function_exists( 'wp_create_term' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/taxonomy.php';
```

Comment:
Wouldn't it be cleaner to simply use `wp_insert_term`, and check for term existence if needed, instead of loading all the rest of the admin-specific code?

This is the implementation of wp_create_term: 

```php
function wp_create_term( $tag_name, $taxonomy = 'post_tag' ) {
	$id = term_exists( $tag_name, $taxonomy );
	if ( $id ) {
		return $id;
	}

	return wp_insert_term( $tag_name, $taxonomy );
}
```

The logic could be adjusted to,

```php
	if ( ! is_a( $main_locale, WP_Term::class ) ) {
		$term_id = term_exists( $tag_name, $taxonomy );
		if ( empty( $term_id ) ) {
			$result = wp_insert_term( wmf_get_current_content_language_slug(), 'content-language' );
			if ( is_wp_error( $result ) ) {
				return null;
			}
			$term_id = $result['term_id'];
		}
		return get_term( $term_id, 'content-language' );		
	}
```

and we wouldn't need anything from the admin taxonomy.php.

Reply:
Yeah, and actually we don't even need the `term_exists` check because that's already handled inside `wmf_get_current_content_language_term()`: https://github.com/wikimedia/shiro-wordpress-theme/blob/7df2deaf3a9faf354dc537f41d6ded11545e1e2a/inc/taxonomy-content-language.php#L57-L60